mm/2s2h/BenGui/Menu.cpp

```diff
@@ -22,66 +24,101 @@ extern PlayState* gPlayState;
-//                         { .tooltip = "Adjusts the culling planes to account for widescreen resolutions. "
-//                                      "This may have unintended side effects." });
-
-// if (gPlayState != NULL) {
-//     ImGui::Separator();
-//     SearchMenuGetItem(MENU_ITEM_FRAME_ADVANCE_ENABLE);
-//     if (gPlayState->frameAdvCtx.enabled) {
-//         SearchMenuGetItem(MENU_ITEM_FRAME_ADVANCE_SINGLE);
-//         SearchMenuGetItem(MENU_ITEM_FRAME_ADVANCE_HOLD);
-//         if (ImGui::IsItemActive()) {
-//             CVarSetInteger("gDeveloperTools.FrameAdvanceTick", 1);
-//         }
-//     }
-// }
-// ImGui::PushStyleColor(ImGuiCol_Button, menuTheme[menuThemeIndex]);
-// RenderWarpPointSection();
-// ImGui::PopStyleColor(1);
-//}
-
-BenMenu::BenMenu(const std::string& consoleVariable, const std::string& name) : GuiWindow(consoleVariable, name) {
+extern std::unordered_map<s16, const char*> warpPointSceneList;
+extern void Warp();
+
+namespace BenGui {}
+
+namespace Ship {
+std::string disabledTempTooltip;
+const char* disabledTooltip;
+bool disabledValue = false;
+
+bool operator==(Color_RGB8 const& l, Color_RGB8 const& r) noexcept {
+    return l.r == r.r && l.g == r.g && l.b == r.b;
+}
+
+bool operator==(Color_RGBA8 const& l, Color_RGBA8 const& r) noexcept {
+    return l.r == r.r && l.g == r.g && l.b == r.b && l.a == r.a;
+}
+
+bool operator<(Color_RGB8 const& l, Color_RGB8 const& r) noexcept {
+    return (l.r < r.r && l.g <= r.g && l.b <= r.b) || (l.r <= r.r && l.g < r.g && l.b <= r.b) ||
+           (l.r <= r.r && l.g <= r.g && l.b < r.b);
+}
+
+bool operator<(Color_RGBA8 const& l, Color_RGBA8 const& r) noexcept {
+    return (l.r < r.r && l.g <= r.g && l.b <= r.b && l.a <= r.a) ||
+           (l.r <= r.r && l.g < r.g && l.b <= r.b && l.a <= r.a) ||
+           (l.r <= r.r && l.g <= r.g && l.b < r.b && l.a <= r.a) ||
+           (l.r <= r.r && l.g <= r.g && l.b <= r.b && l.a < r.a);
+}
+
+bool operator>(Color_RGB8 const& l, Color_RGB8 const& r) noexcept {
+    return (l.r > r.r && l.g >= r.g && l.b >= r.b) || (l.r >= r.r && l.g > r.g && l.b >= r.b) ||
+           (l.r >= r.r && l.g >= r.g && l.b > r.b);
+}
+
+bool operator>(Color_RGBA8 const& l, Color_RGBA8 const& r) noexcept {
+    return (l.r > r.r && l.g >= r.g && l.b >= r.b && l.a >= r.a) ||
+           (l.r >= r.r && l.g > r.g && l.b >= r.b && l.a >= r.a) ||
+           (l.r >= r.r && l.g >= r.g && l.b > r.b && l.a >= r.a) ||
+           (l.r >= r.r && l.g >= r.g && l.b >= r.b && l.a > r.a);
 }
 
-void BenMenu::InitElement() {
+uint32_t GetVectorIndexOf(std::vector<std::string>& vector, std::string value) {
+    return std::distance(vector.begin(), std::find(vector.begin(), vector.end(), value));
+}
+
+void Menu::InsertSidebarSearch() {
+    menuEntries[0].sidebars.emplace("Search", searchSidebarEntry);
+    uint32_t curIndex = 0;
+    if (CVarGetString(menuEntries[0].sidebarCvar, "") != "") {
```

Comment:
Comparison to c literal strings are undefined behavior
```suggestion
    if (!Ship_IsCStringEmpty(CVarGetString(menuEntries[0].sidebarCvar, "")) {
```